h/api_client/api_client.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+import requests
+
+
+class APIError(Exception):
+
+    """Raised when a request to the API fails for any reason."""
+
+    pass
+
+
+class ConnectionError(APIError):
+
+    """Raised when a request to the API fails because of a network problem."""
+
+    pass
+
+
+class Timeout(APIError):
+
+    """Raised when a request to the API times out."""
+
+    pass
+
+
+class Client(object):
+
+    """A client for the Hypothesis API.
+
+    Provides simple methods for calling the API over HTTP.
+
+    Note that this class is deliberately decoupled from the Hypothesis Pyramid
+    app and from Pyramid.
+
+    """
+
+    def __init__(self, base_url):
+        """Initialize a new API client.
+
+        :param base_url: The base URL to the Hypothesis API instance to use.
+            For example: ``"http://hypothesis.is/api"``.
+
+        """
+        self.base_url = base_url
+
+    def get(self, path, params=None):
```

Comment:
If we're going to expose the HTTP methods directly like this, any reason we shouldn't just initialize a requests session with the base URL as a client factory? If we're happy to use the requests API, we can just expose it directly.

I'm generally in favor of exposing the guts of things unless the abstraction offers some value.